java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttMessageListener.java

```diff
@@ -0,0 +1,48 @@
+package com.cumulocity.generic.mqtt.client;
+
+import com.cumulocity.generic.mqtt.client.model.GenericMqttMessage;
+import org.java_websocket.framing.CloseFrame;
+
+import java.net.URI;
+
+/**
+ * Implement this interface to handle events from the websocket server after {@link GenericMqttSubscriber#subscribe(GenericMqttMessageListener)}.
+ */
+public interface GenericMqttMessageListener {
+
+    /**
+     * Called after an opening handshake has been performed and the given websocket is ready to be
+     * written on.
+     *
+     * @param serverUri the server URI the websocket is connected to.
+     */
+    void onOpen(URI serverUri);
+
+    /**
```

Comment:
nitpick: Prefer a slight change to 'Callback received upon GenericMqttMessage message events from the remote host. The messages will be acknowledged by the XYZ if the callback completes normally.'

Replace XYZ with either SDK, library code or (maybe best if correct) GenericMqttSubscriber (which I think implements the ack).

Moved received for similar reasons to previous method use of received. 'Events' do not need a verb as they are 'a happening'.

Less certain about changing 'if no exception is raised' to 'if the callback completes normally'. I think my suggestion is more technically correct/safe, but maybe too technical.